src/pacmod3_node.cpp

```diff
@@ -466,11 +472,14 @@ int main(int argc, char *argv[])
       n.advertise<pacmod_msgs::SystemRptBool>("parsed_tx/marker_lamp_rpt", 20);
     ros::Publisher sprayer_rpt_pub =
       n.advertise<pacmod_msgs::SystemRptBool>("parsed_tx/sprayer_rpt", 20);
+    ros::Publisher hazard_lights_rpt_pub =
+      n.advertise<pacmod_msgs::SystemRptBool>("parsed_tx/hazard_lights_rpt", 20);
 
     pub_tx_list.emplace(CruiseControlButtonsRptMsg::CAN_ID, std::move(cruise_control_buttons_rpt_pub));
     pub_tx_list.emplace(EngineBrakeRptMsg::CAN_ID, std::move(engine_brake_rpt_pub));
     pub_tx_list.emplace(MarkerLampRptMsg::CAN_ID, std::move(marker_lamp_rpt_pub));
     pub_tx_list.emplace(SprayerRptMsg::CAN_ID, std::move(sprayer_rpt_pub));
+    pub_tx_list.insert(std::make_pair(HazardLightRptMsg::CAN_ID, hazard_lights_rpt_pub));
```

Comment:
Please use `emplace` instead of `insert`. `insert` does this:

- `std::make_pair` creates a `std::pair`, which allocates memory.
- `std::make_pair` then passes the two arguments to a constructor for the above `std::pair`.
- `insert` then allocates memory in the `map` for a `std::pair`.
- Since the object created by `std::make_pair` is an rvalue, insert uses the move constructor of `std::pair` to create a new `std::pair` in the `map`.
- The temporary object created by `std::make_pair` is destroyed.

While `emplace` does this:

- `emplace` passes the two arguments to a constructor for `std::pair`, which it allocates in the `map`

`emplace` saves allocation of the temporary.